hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -513,16 +575,41 @@ public SerializerAdapter serializerFor(final Object object) {
             serializer = lookupGlobalSerializer(type);
         }
 
+        //6-A good candidate to be serialized via compact serializer
+        if (compactStreamSerializer.isEnabled() && serializer == null) {
```

Comment:
I think we can check for `serializer == null` first to shortcut the if check as I believe it is cheaper. I believe it will be false, most of the time